<launch>

    <arg name="sim_positions" default="$(find bwi_scavenger)/support/simulation.yaml" />
    <arg name="real_positions" default="$(find bwi_scavenger)/support/real.yaml" />

    <arg name="sim_tolerance" default="0.5" />
    <arg name="real_tolerance" default="0.5" />
    <arg name="logging_directory" default="/home/bwi/shiqi/" />

    <node name="scav_hunt" pkg="bwi_scavenger" type="scav_hunt" />

    <!-- assuming 1m/s speed, this cost represents how many seconds needed -->
    <arg name="analyzing_scene_cost" default="8.0" />

    <node name="target_search" pkg="bwi_scavenger" type="target_search" >
        <param name="positions" value="$(arg real_positions)" />
        <param name="tolerance" value="$(arg real_tolerance)" />
    </node>

    <node name="scavenger_vision_server" pkg="bwi_scavenger" type="vision_server" />

    <node name="blue_shirt" pkg="bwi_scavenger" type="blue_shirt" >
        <param name="directory" value="$(arg logging_directory)" />
    </node>

    <node name="object_detection" pkg="bwi_scavenger" type="object_detection" >
        <param name="directory" value="$(arg logging_directory)" />
    </node>

    <node name="white_board" pkg="bwi_scavenger" type="white_board" >
        <param name="directory" value="$(arg logging_directory)" />
    </node>

    <node name="fetch_object" pkg="bwi_scavenger" type="fetch_object" >
        <param name="directory" value="$(arg logging_directory)" />
    </node>

    <include file="$(find bwi_kr_execution)/launch/bwi_kr_execution.launch" />

</launch>
